tests/components/control4/test_config_flow.py

```diff
@@ -0,0 +1,198 @@
+"""Test the Control4 config flow."""
+import datetime
+
+from pyControl4.account import C4Account
+from pyControl4.director import C4Director
+from pyControl4.error_handling import Unauthorized
+
+from homeassistant import config_entries, setup
+from homeassistant.components.control4.const import DEFAULT_SCAN_INTERVAL, DOMAIN
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+
+from tests.async_mock import AsyncMock, patch
+from tests.common import MockConfigEntry
+
+
+def _get_mock_c4_account(
+    getAccountControllers={
+        "controllerCommonName": "control4_model_00AA00AA00AA",
+        "href": "https://apis.control4.com/account/v3/rest/accounts/000000",
+        "name": "Name",
+    },
+    getDirectorBearerToken={
+        "token": "token",
+        "token_expiration": datetime.datetime(2020, 7, 15, 13, 50, 15, 26940),
+    },
+):
+    c4_account_mock = AsyncMock(C4Account)
+
+    c4_account_mock.getAccountControllers.return_value = getAccountControllers
+    c4_account_mock.getDirectorBearerToken.return_value = getDirectorBearerToken
+
+    return c4_account_mock
+
+
+def _get_mock_c4_director(getAllItemInfo={}):
+    c4_director_mock = AsyncMock(C4Director)
+    c4_director_mock.getAllItemInfo.return_value = getAllItemInfo
+
+    return c4_director_mock
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)